packer/plugin.go

```diff
@@ -24,6 +24,11 @@ type PluginConfig struct {
 	Provisioners       ProvisionerSet
 	PostProcessors     PostProcessorSet
 	DataSources        DatasourceSet
+
+	BuilderRedirects       map[string]string
```

Comment:
I think we should add all of the plugins redirect at once and release with the complete map even when we haven't extracted a plugin yet. This is to make a plugin implicitly available even if it was extracted after this was released. 
I know this will require some error handling for the case the redirect doesn't exist yet. Maybe we can ignore it and assume is still in the core, since they all are maintained by us.

Reply:
Oh, so you mean we implicitly require plugins that we know we will move out; but ignore '404 errors' for these _implicit_ plugins that could not be installed yet. I actually love this suggestion. Looking at the implications.

Reply:
Yep. Because then this will be always compatible. Once we break out a plugin we will start using and printing the warning for that one even if the user doesn't change the Packer version. 
This is going to be good to encourage users to update their templates as soon we have a release available for a plugin.

I like it too!

Reply:
Yeah, I think I like this. The only problem I can think of is if someone is trying to use packer-on-master with all the mappings, while we're first creating the repo. It may exist, but not be quite complete, which could cause weird errors. We'd need to be careful not to push repositories until they're actually ready for use.

Reply:
Packer will fail if it doesn't find a release tag, right? We could ignore these errors as well, in case the repo is there but not released yet.

Reply:
Okay; after some internal convos, I edited the map to be commented values, that we will uncomment when moving stuff out.

Reply:
Nice. Sounds good!